### ISSUE TITLE
Undefined reference to lp::lp_solver::get_column_name when building on FreeBSD 10.3

### ISSUE THREAD

**Issue by arrowd** [Z3Prover/z3#1444]
I'm compiling z3 4.6.0 release on FreeBSD 10.3 and getting the following error:
```

shell/lp_frontend.o:(.data.rel.ro._ZTVN2lp17lp_primal_simplexIddEE[_ZTVN2lp17lp_primal_simplexIddEE]+0x10): undefined reference to `lp::lp_solver<double, double>::get_column_name(unsigned int) const'
shell/lp_frontend.o:(.data.rel.ro._ZTVN2lp9lp_solverIddEE[_ZTVN2lp9lp_solverIddEE]+0x10): undefined reference to `lp::lp_solver<double, double>::get_column_name(unsigned int) const'
util/lp/lp.a(lp_primal_simplex_instances.o):(.data.rel.ro._ZTVN2lp17lp_primal_simplexI8rationalS1_EE[_ZTVN2lp17lp_primal_simplexI8rationalS1_EE]+0x10): undefined reference to `lp::lp_solver<rational, rational>::get_column_name(unsigned int) const'
util/lp/lp.a(lp_solver_instances.o):(.data.rel.ro._ZTVN2lp9lp_solverI8rationalS1_EE[_ZTVN2lp9lp_solverI8rationalS1_EE]+0x10): undefined reference to `lp::lp_solver<rational, rational>::get_column_name(unsigned int) const'
c++: error: linker command failed with exit code 1 (use -v to see invocation)
```

The full build log is here: https://poudriere.mouf.net/lrrr/poudriere/data/103-amd64-default/2018-01-04_14h21m21s/logs/errors/z3-4.6.0.log

Building the same way on FreeBSD 11.1 works OK.

**Comment by NikolajBjorner** [comment]
did the check-in fix this?

**Comment by arrowd** [comment]
I take it, only ` src/util/lp/lp_solver_instances.cpp` change is relevant for my bug? I pulled that in and

> 
shell/lp_frontend.o:(.data.rel.ro._ZTVN2lp17lp_primal_simplexIddEE[_ZTVN2lp17lp_primal_simplexIddEE]+0x10): undefined reference to `lp::lp_solver<double, double>::get_column_name(unsigned int) const'
shell/lp_frontend.o:(.data.rel.ro._ZTVN2lp9lp_solverIddEE[_ZTVN2lp9lp_solverIddEE]+0x10): undefined reference to `lp::lp_solver<double, double>::get_column_name(unsigned int) const'

are gone, but there are still

>util/lp/lp.a(lp_primal_simplex_instances.o):(.rodata._ZTVN2lp17lp_primal_simplexI8rationalS1_EE[_ZTVN2lp17lp_primal_simplexI8rationalS1_EE]+0x8): undefined reference to `lp::lp_solver<rational, rational>::get_column_name(unsigned int) const'
util/lp/lp.a(lp_solver_instances.o):(.rodata._ZTVN2lp9lp_solverI8rationalS1_EE[_ZTVN2lp9lp_solverI8rationalS1_EE]+0x8): undefined reference to `lp::lp_solver<rational, rational>::get_column_name(unsigned int) const'

errors.

**Comment by NikolajBjorner** [comment]
@levnach. Can you check?

**Comment by NikolajBjorner** [comment]
@levnach. Can you check?

**Comment by arrowd** [comment]
This is how I made it work locally: https://svnweb.freebsd.org/ports/head/math/z3/files/patch-src_util_lp_lp__solver__instances.cpp?view=markup

**Comment by NikolajBjorner** [comment]
This was fixed in a5caa506067a949afc445c5bf467fe8403538ec9
